tiatoolbox/utils/visualization.py

```diff
@@ -0,0 +1,122 @@
+# ***** BEGIN GPL LICENSE BLOCK *****
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# The Original Code is Copyright (C) 2021, TIALab, University of Warwick
+# All rights reserved.
+# ***** END GPL LICENSE BLOCK *****
+
+"""Visualisation and overlay functions used in tiatoolbox."""
+
+import numpy as np
+import cv2
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+
+
+def overlay_patch_prediction(
+            img : np.ndarray,
+            prediction : np.ndarray,
+            alpha : float = 0.35,
+            label_info : dict = None,
+            ax=None,
+        ):
+    """Generate an overlay, given a 2D prediction map.
+
+    Args:
+        img (ndarray): Image to overlay the results on top of.
+        prediction (ndarray): 2D prediction map. Multi-class prediction should have
+            values ranging from 0 to N-1, where N is the number of classes.
+        label_info (dict): A dictionary contains the mapping for each integer value
+            within `prediction` to its string and color. [int] : (str, (int, int, int)).
+            By default, integer will be taken as label and color will be random.
+        alpha (float): Opacity value used for the overlay.
+        ax (ax): Matplotlib ax object.
+    """
+    if img.shape[:2] != prediction.shape[:2]:
+        raise ValueError(
+            'Mismatch shape `img` {0} vs `prediction` {1}.'.format(
+                img.shape[:2], prediction.shape[:2]
+            ))
+
+    img = img.astype("uint8")
```

Comment:
May be check if the image is scaled between 0 to 255. If it is in float format then change it to uint8.